content/en/docs/components/pipelines/sdk/v2/component-development.md

```diff
@@ -0,0 +1,606 @@
+*   [`kfp.components.load_component_from_text`][kfp-load-comp-text]:
+    Use this method to load your component specification YAML from a string.
+    This method is useful for rapidly iterating on your component
+    specification.
+
+These functions create a factory function that you can use to create
+[`ContainerOp`][kfp-containerop] instances to use as steps in your pipeline.
+This factory function's input arguments include your component's inputs and
+the paths to your component's outputs. The function signature may be modified
+in the following ways to ensure that it is valid and Pythonic.
+
+*   Inputs with default values will come after the inputs without default
+    values and outputs.
+*   Input and output names are converted to Pythonic names (spaces and symbols
+    are replaced with underscores and letters are converted to lowercase). For
+    example, an input named `Input 1` is converted to `input_1`. 
+
+The following example demonstrates how to load the text of your component
+specification and run it in a single-step pipeline. Before you run this
+example, update the component specification to use the component
+specification you defined in the previous sections.
+
+```python
+import kfp
+import kfp.components as comp
+
+create_step_get_lines = comp.load_component_from_text("""
+name: Get Lines
+description: Gets the specified number of lines from the input file.
+
+inputs:
+- {name: Input 1, type: String, description: 'Data for input 1'}
+- {name: Parameter 1, type: Integer, default: '100', description: 'Number of lines to copy'}
+
+outputs:
+- {name: Output 1, type: String, description: 'Output 1 data.'}
+
+implementation:
+  container:
+    image: gcr.io/my-org/my-image@sha256:a172..752f
+    # command is a list of strings (command-line arguments). 
+    # The YAML language has two syntaxes for lists and you can use either of them. 
+    # Here we use the "flow syntax" - comma-separated strings inside square brackets.
+    command: [
+      python3, 
+      # Path of the program inside the container
+      /pipelines/component/src/program.py,
+      --input1-path,
+      {inputPath: Input 1},
+      --param1, 
+      {inputValue: Parameter 1},
+      --output1-path, 
+      {outputPath: Output 1},
+    ]""")
+
+# create_step_get_lines is a "factory function" that accepts the arguments
+# for the component's inputs and output paths and returns a pipeline step
+# (ContainerOp instance).
+#
+# To inspect the get_lines_op function in Jupyter Notebook, enter 
+# "get_lines_op(" in a cell and press Shift+Tab.
+# You can also get help by entering `help(get_lines_op)`, `get_lines_op?`,
+# or `get_lines_op??`.
+
+# Define your pipeline
+@dsl.pipeline(
+    pipeline_root='gs://my-pipeline-root/example-pipeline',
+    name="example-pipeline",
+) 
+def my_pipeline():
+    get_lines_step = create_step_get_lines(
+        # Input name "Input 1" is converted to pythonic parameter name "input_1"
```

Comment:
nit: Maybe it's only me, but I always feel itchy about such magic conversion. I wonder if we can just use `input_1` in the component yaml, which would be a legal identifier in Python (or potentially any other common programming languages), and requires no conversion.

Reply:
Done.